dvc/project.py

```diff
@@ -647,6 +647,10 @@ def _collect_used_cache(self,
                             force=False,
                             jobs=None):
         if not out.use_cache or not out.info:
+            if not out.info:
+                logger.warning("Local out: {} has no info attached. It will "
```

Comment:
How about something like "Output '{}'(stage '{}') is missing version info. Cache for it will not be collected. Use `dvc repro` to get your pipeline up to date."?